[Svelte] Warning / suggested configuration when installing via composer

Installing ux-svelte via `composer require symfony/ux-svelte` have warning / missing webpack config entry. I'm not sure though if this is related to Svelte or UX, since it does not export the path `./internal` when adding `conditionNames` to the Webpack configuration.

## The problem

- Create a new project by `symfony new --webapp my-app`
- Add Svelte to project `composer require symfony/ux-svelte`
- Build dependencies using yarn `yarn install` && `yarn build`

Dependencies added and installed:
```json
{
    "devDependencies": {
        ...
        "@symfony/stimulus-bridge": "^3.2.0",
        "@symfony/ux-svelte": "file:vendor/symfony/ux-svelte/assets",
        ...
        "svelte": "^3.0",
        "svelte-loader": "^3.1.9",
        ...
    },
   ...
}
```

This produces a warning, but still build successfully:
```bash
WARNING: You should add "svelte" to the "resolve.conditionNames" array in your webpack config. See https://github.com/sveltejs/svelte-loader#resolveconditionnames for more information
```

Attempting to fix the warning by changing default webpack configuration:
```js
let config = Encore.getWebpackConfig();
config.resolve.conditionNames = ["svelte"];
module.exports = config;
```

This will produce, if attempting to build again by `yarn build`, the following error message:
```bash
ERROR  Failed to compile with 1 errors                                                                                                                                                                                                   
error  in ./assets/svelte/controllers/Hello.svelte                                                                                                                                                                                       

Module not found: Error: Package path ./internal is not exported from package ./node_modules/svelte (see exports field in ./node_modules/svelte/package.json)
```

Which can be fixed by updating Svelte to ^4.0 in `packages.json`, this is however probably a "solution" unless `ux-svelte` can increment Svelte version without breaks:
```json
{
    "devDependencies": {
        ...
        "svelte": "^3.0",
        ...
    },
   ...
}
```

With a single dependency warning from ux-svelte:
```bash
warning " > @symfony/ux-svelte@1.0.0" has incorrect peer dependency "svelte@^3.0".
```

Hello! Thanks for the detailed issue :)

A)
Here is the migration guide: https://svelte.dev/docs/v4-migration-guide
There are no BC breaks for us! 🎉 

It's recommended to replace the type `SvelteComponent` with `SvelteComponent<any>` in the typescript files.
I'll open a PR for that + allow both versions in the package.json of ux-svelte

B)
I did some tests: `config.resolve.conditionNames = ["svelte"];` works with both v3 et v4 👍
 
But to get rid of the `Module not found: Error: Package path ./internal is not exported from package ./node_modules/svelte (see exports field in ./node_modules/svelte/package.json)` error in v3, you need to add another config line to webpack:

`config.resolve.alias.svelte = path.resolve('node_modules', 'svelte')`

But adding this line doesn't work with v4 so we have to add it conditionally

This is related to the webpack svelte-loader : https://github.com/sveltejs/svelte-loader/issues/234#issuecomment-1607058996

--- a/src/Svelte/assets/package.json
+++ b/src/Svelte/assets/package.json
@@ -21,12 +21,12 @@
     },
     "peerDependencies": {
         "@hotwired/stimulus": "^3.0.0",
-        "svelte": "^3.0"
+        "svelte": "^3.0 || ^4.0"
     },
     "devDependencies": {
         "@hotwired/stimulus": "^3.0.0",
         "@sveltejs/vite-plugin-svelte": "^2.4.6",
         "@types/webpack-env": "^1.16",
-        "svelte": "^3.0"
+        "svelte": "^3.0 || ^4.0"
     }
 }

--- a/src/Svelte/assets/src/loader.ts
+++ b/src/Svelte/assets/src/loader.ts
@@ -13,10 +13,10 @@ import type { SvelteComponent } from 'svelte';
 import { ComponentCollection, components } from './components.js';
 
 declare global {
-    function resolveSvelteComponent(name: string): typeof SvelteComponent;
+    function resolveSvelteComponent(name: string): typeof SvelteComponent<any>;
 
     interface Window {
-        resolveSvelteComponent(name: string): typeof SvelteComponent;
+        resolveSvelteComponent(name: string): typeof SvelteComponent<any>;
     }
 }
 

--- a/src/Svelte/assets/src/register_controller.ts
+++ b/src/Svelte/assets/src/register_controller.ts
@@ -12,10 +12,10 @@
 import type { SvelteComponent } from 'svelte';
 
 declare global {
-    function resolveSvelteComponent(name: string): typeof SvelteComponent;
+    function resolveSvelteComponent(name: string): typeof SvelteComponent<any>;
 
     interface Window {
-        resolveSvelteComponent(name: string): typeof SvelteComponent;
+        resolveSvelteComponent(name: string): typeof SvelteComponent<any>;
     }
 }
 

--- a/src/Svelte/assets/test/fixtures/MyComponent.svelte
+++ b/src/Svelte/assets/test/fixtures/MyComponent.svelte
@@ -3,6 +3,6 @@
     export let name = "without props";
 </script>
 
-<div transition:fade={{ duration: 100 }}>
+<div transition:fade|global={{ duration: 100 }}>
     <div>Hello {name}</div>
 </div>

--- a/src/Svelte/doc/index.rst
+++ b/src/Svelte/doc/index.rst
@@ -8,7 +8,7 @@ Svelte is a JavaScript framework for building user interfaces.
 Symfony UX Svelte provides tools to render Svelte components from Twig,
 handling rendering and data transfers.
 
-Symfony UX Svelte supports Svelte 3 only.
+Symfony UX Svelte supports Svelte 3 and Svelte 4.
 
 Installation
 ------------
@@ -41,10 +41,10 @@ Next, install a package to help Svelte:
 
 .. code-block:: terminal
 
-    $ npm install svelte svelte-loader --save-dev
+    $ npm install svelte-loader --save-dev
 
     # or use yarn
-    $ yarn add svelte svelte-loader --dev
+    $ yarn add svelte-loader --dev
 
 That's it! Any files inside ``assets/svelte/controllers/`` can now be rendered as
 Svelte components.
@@ -97,7 +97,7 @@ Svelte client-side component API:
         export let name = "Svelte";
     </script>
 
-    <div transition:fade>Hello {name}</div>
+    <div transition:fade|global>Hello {name}</div>
 
 
 .. code-block:: html+twig

